### DISCUSSION TITLE
Where is the guide for dummies?

### DISCUSSION THREAD

**Discussion by SSmale** [devcontainers/community#91] [Q&A] [answered]
I seem to have missed the guide for dummies on using dev containers. they seem really cool, but I seem to spend more time fiddling and finding my way in the dark than actually using them.

I am using VSCode to launch them is that makes any difference to the below things I have not yet found:
- 9 times out of 10 when I add a feature, it seems to fail to build.
    - I have tried to add maven to the java build, seems it needs to be in the java image config not as a feature 🤷‍♂️
    - I have just tried to add ansible to an alpine image (also tried to add python too to install ansible via). both failed
- if you get vscode to create a dev container without first saving the config file and it fails etc it seems to eat it.
- there is a k8,docker,minikube container that looks like it should be launched as a secondary container but that does not seem to be a thing
- can you get it to launch other containers alongside? If I would like a DB or RabbitMQ container to talk to but separately.
- can I have a service running that I talk to in another Docker container (like Vault)? how does the comms work there?
- they take an age to launch compared to 'normal' docker containers
- I have yet to find a guide on best practices on using them
- mounting vs cloning git repos....

This maybe an issue with my other knowledge too (like docker) which is where a 'here dummy, step 1' guide would be super helpful!

**Comment by eljog** [comment, marked as answer]
Thank you so much for writing a detailed issue, we appreciate it.
If you haven't already explored https://containers.dev/ has the comprehensive documentation and [guides](https://containers.dev/guides) for devcontainers and features.

> can you get it to launch other containers alongside? If I would like a DB or RabbitMQ container to talk to but separately.
can I have a service running that I talk to in another Docker container (like Vault)? how does the comms work there?

Docker Compose is a great way to define a multi-container development environment. Rather than adding things like databases or redis to your Dockerfile, you can reference existing images for these services and focus your Dev Container’s content on tools and utilities you need for development. More details can be found in the **[guid here](https://containers.dev/guide/dockerfile#docker-compose)**.

Hope that helps. Thank you for trying out devcontainers.
